### PR TITLE
add arq integration; add qualname to wrap_injection

### PR DESCRIPTION
**Pull request by shalmeo**

### CODE REVIEW

File: src/dishka/integrations/arq.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+DISHKA_CONTAINER_KEY: Final = "dishka_container"
+
+logger = logging.getLogger(__name__)
+
+
+def inject(func):
+    return wrap_injection(
+        func=func,
+        remove_depends=True,
+        container_getter=lambda p, _: p[0][DISHKA_CONTAINER_KEY],
```
Comment (Tishka17) [inline src/dishka/integrations/arq.py:23]: is it always 2 arguements?

File: examples/arq/worker.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+async def main():
+    logging.basicConfig(
+        level=logging.DEBUG,
+        format="%(asctime)s  %(process)-7s %(module)-20s %(message)s",
+    )
+
+    worker = Worker(functions=[get_content], redis_settings=RedisSettings())  # type: ignore
+
+    container = make_async_container(GatewayProvider())
+    setup_dishka(container=container, worker=worker)
```
Comment (Tishka17) [inline examples/arq/worker.py:45]: As I see main way to start arq worker is to run `arq myfile.WorkerSettings`. We need support it

File: src/dishka/integrations/arq.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+async def job_end(context: dict[Any, Any]) -> None:
+    sub_container: AsyncContainer = context[DISHKA_CONTAINER_KEY]
+    await sub_container.close()
+
+
+def setup_dishka(container: AsyncContainer, worker: Worker) -> None:
+    if any((worker.on_job_start, worker.on_job_end)):
+        logger.warning("Dishka setup will override existing worker hooks")
```
[the remark this answers is not in the document]
Comment (shalmeo) [reply src/dishka/integrations/arq.py:41]: What do you mean?

File: src/dishka/integrations/arq.py (hunk 1 above)
Comment (shalmeo) [reply src/dishka/integrations/arq.py:23]: Yes

File: examples/arq/worker.py (hunk 2 above)
Comment (shalmeo) [reply examples/arq/worker.py:45]: what will the signature look like?

setup_dishka(*providers, worker: Worker | None = None, worker_settings: WorkerSettingsType | None = None)?

File: src/dishka/integrations/arq.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tishka17) [reply src/dishka/integrations/arq.py:41]: if there is on_job_start previously set, we can save it and call from our function

**Pull request merged**